DuckDuckGo/FeedbackViewController.swift

```diff
@@ -0,0 +1,179 @@
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+
+import Foundation
+
+import UIKit
+import Core
+import ToastSwiftFramework
+
+class FeedbackViewController: UIViewController {
+    
+    private struct ViewConstants {
+        static let urlTextHeight: CGFloat = 38
+        static let urlTextPadding: CGFloat = 4
+    }
+    
+    @IBOutlet weak var scrollView: UIScrollView!
+    @IBOutlet weak var brokenSiteSwitch: UISwitch!
+    @IBOutlet weak var urlTextFieldHeight: NSLayoutConstraint!
+    @IBOutlet weak var urlTextField: UITextField!
+    @IBOutlet weak var messageTextView: UITextView!
+    @IBOutlet weak var messagePlaceholderText: UILabel!
+    @IBOutlet weak var submitButton: UIButton!
+    
+    private var feedbackModel = FeedbackModel()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.blur(style: .dark)
+        loadModel()
+        configureViews()
+        registerForKeyboardNotifications()
+        refreshMode()
+    }
+    
+    func prepareForSegue(isBrokenSite: Bool, url: String?) {
+        feedbackModel.isBrokenSite = isBrokenSite
+        feedbackModel.url = url
+    }
+    
+    private func loadModel() {
+        brokenSiteSwitch.isOn = feedbackModel.isBrokenSite
+        messageTextView.text = feedbackModel.message
+        urlTextField.text = feedbackModel.url
+    }
+    
+    private func configureViews() {
+        messageTextView.delegate = self
+        messageTextView.layer.borderWidth = 1
+        messageTextView.layer.borderColor = UIColor.mercury.cgColor
+        urlTextField.layer.borderWidth = 1
+        urlTextField.layer.borderColor = UIColor.mercury.cgColor
+        urlTextField.layer.sublayerTransform = CATransform3DMakeTranslation(ViewConstants.urlTextPadding, 0, 0)
+    }
+    
+    private func registerForKeyboardNotifications() {
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(keyboardDidShow),
+                                               name: NSNotification.Name.UIKeyboardDidShow,
+                                               object: nil)
+        
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(keyboardWillHide),
+                                               name: NSNotification.Name.UIKeyboardWillHide,
+                                               object: nil)
+    }
+    
+    @objc private func keyboardDidShow(notification: NSNotification) {
```

Comment:
Not sure if this is working as desired. On iPhone 8 Plus and iPhone SE simulators there's no movement of the field so it leaves a small input area and you have to manually scroll up to submit.  So it feels a bit wrong.

 e.g. 

![simulator screen shot - iphone se - 2018-06-27 at 10 30 27](https://user-images.githubusercontent.com/81197/41966089-2340912e-79f6-11e8-8e03-f31d444c5d8e.png)

However, I expected it to scroll up like this:

![simulator screen shot - iphone se - 2018-06-27 at 10 30 37](https://user-images.githubusercontent.com/81197/41966100-3166fd9c-79f6-11e8-9e01-53305ce55430.png)

Reply:
Another example: 
![simulator screen shot - iphone se - 2018-06-27 at 10 44 50](https://user-images.githubusercontent.com/81197/41966510-378fe1a6-79f7-11e8-9b80-8da0b4ba82de.png)

Reply:
@brindy and I discussed this and it is sadly the default behavior to move the field just enough so that typing can happen. We will run with it and if we get feedback we will revisit.